includes/Infrastructure/WooCommerce/Adapter/OrderAdapter.php

```diff
@@ -0,0 +1,177 @@
+	 * @return WC_Order
+	 */
+	public function get_wc_order(): WC_Order {
+		return $this->wc_order;
+	}
+
+	/**
+	 * Get the lines of the order.
+	 * This method retrieves the items from the order and maps them to OrderLineInterface.
+	 *
+	 * @return OrderLineAdapter[]
+	 */
+	public function getOrderLines(): array {
+		return array_map(
+			fn( $item ) => new OrderLineAdapter( $item ),
+			$this->wc_order->get_items()
+		);
+	}
+
+	/**
+	 * Get the Payment identifier.
+	 * This method retrieves the transaction ID of the order, which is used to identify the payment
+	 *
+	 * @return string The transaction ID of the order.
+	 */
+	public function getPaymentId(): string {
+		return $this->wc_order->get_transaction_id();
+	}
+
+	/**
+	 * Get the merchant reference.
+	 * This method retrieves the order number, which serves as the merchant reference for the order.
+	 *
+	 * @return string The order number of the order.
+	 */
+	public function getMerchantReference(): string {
+		return $this->wc_order->get_order_number();
+	}
+
+	/**
+	 * Get the remaining refund amount.
+	 * This method calculates the remaining refund amount by subtracting the total refunded amount from
+	 * the total order amount.
+	 *
+	 * @return float The remaining refund amount.
+	 */
+	public function getRemainingRefundAmount(): float {
+		return $this->get_total() - $this->get_total_refunded();
+	}
+
+	/**
+	 * Check if the order is fully refunded.
+	 * This method checks if the remaining refund amount is zero, indicating that the order has been
+	 * fully refunded.
+	 *
+	 * @return bool True if the order is fully refunded, false otherwise.
+	 */
+	public function isFullyRefunded(): bool {
+		return round( $this->getRemainingRefundAmount(), 3 ) === round( 0, 3 );
+	}
+
+	/**
+	 * Check if the order is paid with Alma payment method.
+	 * This method checks if the order's payment method is one of the Alma payment methods.
+	 *
+	 * @return bool True if the order is paid with Alma, false otherwise.
+	 */
+	public function isPaidWithAlma(): bool {
+		$gateways = WC()->payment_gateways()->payment_gateways();
+		$gateway  = $gateways[ $this->wc_order->get_payment_method() ] ?? null;
+
+		if ( $gateway instanceof AbstractGateway ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if the order has a transaction ID.
+	 * This method checks if the order has a transaction ID set, which is necessary for processing
+	 * refunds and other payment-related operations.
+	 *
+	 * @return bool True if the order has a transaction ID, false otherwise.
+	 */
+	public function hasATransactionId(): bool {
+		if ( ! $this->wc_order->get_transaction_id() ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if the order is refundable.
+	 * This method checks if the order is paid with Alma and has a transaction ID,
+	 * indicating that it can be refunded.
+	 *
+	 * @return bool True if the order is refundable, false otherwise.
+	 */
+	public function isRefundable(): bool {
```

Comment:
Function need to be rename because, just with the name, we don't know we check the order was paid with Alma